src/librustc_save_analysis/lib.rs

```diff
@@ -1141,10 +1141,15 @@ fn find_config(supplied: Option<Config>) -> Config {
     if let Some(config) = supplied {
         return config;
     }
+
     match env::var_os("RUST_SAVE_ANALYSIS_CONFIG") {
-        Some(config_string) => rustc_serialize::json::decode(config_string.to_str().unwrap())
-            .expect("Could not deserialize save-analysis config"),
         None => Config::default(),
+        Some(config) => config.to_str()
+            .ok_or(())
+            .map_err(|_| error!("`RUST_SAVE_ANALYSIS_CONFIG` isn't UTF-8"))
```

Comment:
Non-UTF8 error is now handled and if any error occurs, we now use default config instead of panicking here

Reply:
Can we add a test for this?

Reply:
Added in https://github.com/rust-lang/rust/pull/60053/commits/b03123426c60dc22d84291a0fcecef67451514a8